next/components/pages/SearchPage.tsx

```diff
@@ -158,33 +158,36 @@ const SearchPage = () => {
                 {t('resultsFound', { count: 0 })}
               </motion.div>
             ) : (
-              <div ref={resultsRef} className="flex flex-col">
+              <div ref={resultsRef} className="flex flex-col p-1">
                 {data?.hits.map(({ title, type, id, slug }, index) => {
                   const link = getPathForEntity(
                     type === 'page' ? { type, id: String(id) } : { type, slug }
                   )
 
                   return (
-                    // eslint-disable-next-line react/no-array-index-key
-                    <Link key={index} href={link ?? '#'}>
-                      <div className="group flex items-center justify-between gap-x-6 border-b border-border-dark bg-white py-4">
-                        <div className="flex grow flex-col gap-y-2">
-                          <div className="flex items-start gap-x-4 max-md:justify-between">
+                    <div
+                      // eslint-disable-next-line react/no-array-index-key
+                      key={index}
+                      className="group flex items-center justify-between gap-x-6 border-b border-border-dark bg-white py-4"
+                    >
+                      <div className="flex grow flex-col gap-y-2">
+                        <div className="flex items-start gap-x-4 max-md:justify-between">
+                          <MLink href={link ?? '#'} variant="basic" stretched>
                             <h2>{title}</h2>
-                            {/* py set to 2px to coun also with the border */}
-                            <span className="flex h-6 shrink-0 items-center rounded-[4px] border border-dark px-2 py-[3px] text-[12px] leading-[18px]">
-                              {/* TODO proper translation keys */}
-                              {t(`searchTags.${type}`)}
-                            </span>
-                          </div>
-                          <div className="flex items-center text-xs text-foreground-body">
-                            <span>{link}</span>
-                          </div>
+                          </MLink>
```

Comment:
This is good. But as I mentioned in comment in the issue, you must use `position: relative` on the topmost component of the "result row", so the link is stretched only on the result row div, not the whole screen. Relative position creates some kind out boundaries, where the link can stretch (see `MLink` component what the `stretched` actually does).

![image](https://github.com/bratislava/mestskakniznica.sk/assets/19418224/2113db0f-b8fd-47af-865f-81a8b93cf30e)

![image](https://github.com/bratislava/mestskakniznica.sk/assets/19418224/c3d16f2c-8120-488a-bca7-017ca6a8576a)